ocaml-eglot.el

```diff
@@ -771,7 +779,10 @@ OCaml Eglot provides standard implementations of the various custom-requests
   :lighter " OCaml-eglot"
   :keymap ocaml-eglot-map
   :group 'ocaml-eglot
-  (add-hook 'find-file-hook #'ocaml-eglot--file-hook))
+  (add-hook 'find-file-hook #'ocaml-eglot--file-hook)
+  (if ocaml-eglot
+      (add-hook 'eglot-managed-mode-hook #'ocaml-eglot--enable-xref-backend nil t)
+    (remove-hook 'eglot-managed-mode-hook #'ocaml-eglot--enable-xref-backend t)))
 
```

Comment:
Does this work right if eglot-managed-mode was already enabled before ocaml-eglot-mode?

This seems a bit complex.  Our minor mode depends on another minor mode being enabled, so normally we would just enable that other minor mode here.  But because what we're actually depending on is eglot and eglot-managed-mode, we can't just do (eglot--managed-mode 1) here...

@joaotavora Do you have any suggestions on how a minor mode which depends on eglot should be initialized?  Should the minor mode maybe fail if eglot isn't enabled, or maybe it should forcibly enable eglot, or maybe something else?